Hydra/WebSocket/WebSocket.cs

```diff
@@ -64,7 +64,29 @@ internal WebSocket(Socket socket, PipeReader reader, PipeWriter writer, int back
             CancellationToken = cancellationToken;
         }
 
-        public static HttpResponse Response(Server.WebSocketHandler handler) => new WebSocketResponse(handler);
+        /// <summary>
+        /// Creates a response suitable for upgrading a request to a <see cref="WebSocket"/>.
+        /// </summary>
+        /// <param name="request">HTTP request to validate the WSS headers with</param>
+        /// <param name="handler">Delegate which will process the <see cref="WebSocket"/> once the handshake
+        /// is complete</param>
+        /// <returns>An HTTP response for the upgrade, or alternatively a code 400 response any of the necessary
+        /// WSS headers are missing or incorrect</returns>
+        public static HttpResponse Response(HttpRequest request, Server.WebSocketHandler handler)
+        {
+            // Verify all of the WebSocket headers
+            if (!request.Headers.TryGetValue("Sec-WebSocket-Key", out var clientKey)
+                || (request.Headers.TryGetValue("Connection", out var conn)
+                    && conn != "Upgrade") 
+                || (request.Headers.TryGetValue("Upgrade", out var upgrade)
+                    && upgrade != "websocket")
+                || (request.Headers.TryGetValue("Sec-WebSocket-Version", out var webSocketVersion)
+                    && webSocketVersion != "13")) // v13 is the only supported version by Hydra
+                return new HttpResponse(400);
```

Comment:
The spec is actually pretty specific as to what kind of response has to be sent depending on which headers are not valid in the request, it's not necessarily a 400.

Reply:
Ah I didn't know this, I will fix.

Reply:
The implementation also doesn't support custom extensions, so that should also be checked and rejected according to spec

Reply:
> The spec is actually pretty specific as to what kind of response has to be sent depending on which headers are not valid in the request, it's not necessarily a 400.

Is it?

According to what I can see:
> If
   the server, while reading the handshake, finds that the client did
   not send a handshake that matches the description below (note that as
   per [[RFC2616](https://datatracker.ietf.org/doc/html/rfc2616)], the order of the header fields is not important),
   including but not limited to any violations of the ABNF grammar
   specified for the components of the handshake, the server MUST stop
   processing the client's handshake and return an HTTP response with an
   appropriate error code (such as 400 Bad Request).

Reply:
Oh I misinterpreted that